### PR TITLE
added docs for setting proxy via CF

### LINKED ISSUE #1598: Add support for APIG proxy settings
Give users the option to specify a certain endpoint to act as an HTTP proxy to another URI. This needs some thinking because setting an endpoint as a proxy means that the endpoint is not tied to a specific function, so it wouldn't make sense to have the config inside an `http` event object. Maybe this particular option could be a higher level config? How about:

```yml
service: my-service
config:
    aws:
        proxies:
            - method: get
              path: users/list
              url: www.example.com/users
            - method: get
              path: posts/list
              url: www.example.com/posts
```

Thats one way to do it, but it looks like too much config. One thing to keep in mind is that users can easily create proxy endpoints with CloudFormation if we document it well. They just need to know the correct template to add and how to reference the restApi id and so on.

What you guys think? should we just leave this for custom CF resources? or provide this nice config based abstraction layer? cc/ @flomotlik @pmuens @ac360 

Parent Issue:
[#1408](https://github.com/serverless/serverless/issues/1408)

### PR DESCRIPTION
**Pull request by eahefnawy**
This PR shows how to add custom cloud formation resources by setting up an HTTP proxy as an example. Resolves #1598 and #1603 

cc/ @flomotlik @pmuens @ac360

### CODE REVIEW

File: docs/guide/custom-provider-resources.md [hunk 1]
```diff
@@ -27,6 +27,62 @@ resources:
...
+          Type: HTTP
+          Uri: http://serverless.com # the URL you want to set a proxy to
+          RequestTemplates:
+            application/json: ""
+          IntegrationResponses:
+            StatusCode: 200
+            ResponseParameters: {}
+            ResponseTemplates:
+              application/json: ""
+
```
Comment (pmuens) [inline docs/guide/custom-provider-resources.md]: Can we remove this newline?

File: lib/plugins/aws/deploy/compile/events/apiGateway/README.md [hunk 2]
```diff
@@ -118,4 +118,7 @@ functions:
                     resultTtlInSeconds: 0
                     identitySource: method.request.header.Auth
                     identityValidationExpression: someRegex
-```
\ No newline at end of file
+```
+
+### Setting an HTTP proxy on API Gateway
+Setting an API Gateway proxy can easily be done by adding two custom Cloud Formation resource templates to your `serverless.yaml` file. [Check this guide for more info on how to set up a proxy using custom CloudFormation resources in `serverless.yaml`](https://github.com/serverless/serverless/blob/v1.0/docs/guide/custom-provider-resources.md).
```
Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: `Cloud Formation` --> `CloudFormation`

File: docs/guide/custom-provider-resources.md [hunk 3]
```diff
@@ -27,6 +27,62 @@ resources:
...
+To set up an HTTP proxy, you'll need two CloudFormation templates, one for the endpoint (known as resource in CF), and one for method. These two templates will work together to construct your proxy. So if you want to set `your-app.com/serverless` as a proxy for `serverless.com`, you'll need the following two templates in your `serverless.yaml`:
+
+
+```yaml
+# serverless.yaml
+service: service-name
+provider: aws
+functions:
+  ...
+resources:
```
Comment (pmuens) [inline docs/guide/custom-provider-resources.md]: For better readability I would add a newline before this section.

File: docs/guide/custom-provider-resources.md [hunk 4]
```diff
@@ -27,6 +27,62 @@ resources:
         Key: Value
 ```
 
+### Example custom resources - HTTP Proxy
+As a practical example for adding a custom resource to your service, we're going to demonstrate how you can create an API Gateway HTTP proxy using Cloud Formation templates/resources.
```
Comment (pmuens) [inline docs/guide/custom-provider-resources.md]: `Cloud Formation` --> `CloudFormation`

File: lib/plugins/aws/deploy/compile/events/apiGateway/README.md (hunk 2 above)
Comment (eahefnawy) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: Nice catch! 💥

**Comment by pmuens** [comment]
Great! GTM from my side! :shipit:

**Pull request merged**